modules/shared_image_gallery/image_definitions/Image_definitions.tf

```
@@ -15,6 +15,7 @@ resource "azurerm_shared_image" "image" {
   resource_group_name = var.resource_group_name
   location            = var.location
   os_type             = var.settings.os_type
+  hyper_v_generation  = var.hyper_v_generation
```

```suggestion
  hyper_v_generation  = try(var.settings.hyper_v_generation, null)
```
I would suggest we catch it under the settings object and we put it as optional as it is in the provider: https://registry.terraform.io/providers/hashicorp/azurerm/2.99.0/docs/resources/shared_image#hyper_v_generation